pkg/segment/aggregations/timechartagg.go

```diff
@@ -255,22 +264,37 @@ func MergeVal(eVal *utils.CValueEnclosure, eValToMerge utils.CValueEnclosure, hl
 		fallthrough
 	case utils.Sum:
 		aggFunc = utils.Sum
-		// TODO: should merge values for not numeric agg
-		// case utils.Cardinality:
-		// 	if useAdditionForMerge {
-		// 		aggFunc = utils.Sum
-		// 	} else {
-		// 		log.Error("fjl test1:", hll)
-		// 		log.Error("fjl test2:", hllToMerge)
-		// 		err := hll.Merge(hllToMerge)
-		// 		if err != nil {
-		// 			log.Errorf("MergeVal: failed to merge hyperloglog stats: %v", err)
-		// 		}
-		// 		eVal.CVal = hll.Estimate()
-		// 		eVal.Dtype = utils.SS_DT_UNSIGNED_NUM
-		// 		return
-		// 	}
-		// case utils.Values:
+	// TODO: should merge values for not numeric agg
+	case utils.Cardinality:
+		if useAdditionForMerge {
+			aggFunc = utils.Sum
+		} else {
+			err := hll.Merge(hllToMerge)
+			if err != nil {
+				log.Errorf("MergeVal: failed to merge hyperloglog stats: %v", err)
+			}
+			eVal.CVal = hll.Estimate()
+			eVal.Dtype = utils.SS_DT_UNSIGNED_NUM
+			return
+		}
+	case utils.Values:
+		// Can not do addition for values func
+		if useAdditionForMerge {
+			return
+		}
```

Comment:
Does this need a TODO and/or a log message, or should it just be `return`?

Reply:
It should be return. `useAdditionForMerge` means sum up their values. It is used by `limit` to score single agg. However, for single agg `values`, we can not sum up them, we use `frequency` to score them.